### PR TITLE
consul: 0.7.5 -> 0.8.5

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

Version update

* consul: 0.7.5 -> 0.8.5
* consul-alerts: 0.3.3 -> 0.5.0
* consul-alerts: added explicit --log-level, it fails to start without
* consul service: removed legacy option 'leaveOnStop', since 0.7 it can be set in 'extraConfig'
* consul service: removed insecure and useless option 'dropPrivileges' which allowed to run consul daemon as root

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/networking/consul.nix [hunk 1]
```diff
@@ -97,7 +91,15 @@ in
       };
```
Comment (volth) [inline nixos/modules/services/networking/consul.nix]: Does anyone need this option to run ```consul``` as ```root``` ?

File: nixos/modules/services/networking/consul.nix [hunk 2]
```diff
@@ -240,7 +230,8 @@ in
               --alert-addr=${cfg.alerts.listenAddr} \
               --consul-addr=${cfg.alerts.consulAddr} \
               ${optionalString cfg.alerts.watchChecks "--watch-checks"} \
-              ${optionalString cfg.alerts.watchEvents "--watch-events"}
+              ${optionalString cfg.alerts.watchEvents "--watch-events"} \
+              --log-level=warn
```
Comment (volth) [inline nixos/modules/services/networking/consul.nix]: it just fails to start without explicit ```--log-level=```.
the previous version (before the PR)  too.

**Comment by volth** [comment]
I am not happy with ```restartTriggers``` logic, it would be better to send ```SIGHUP``` to the daemon.

**Comment by joachifm** [comment]
LGTM. One thing to consider is that removing options is a breaking change. For convenience, you could add corresponding `mkRemovedOptionModule`s. (I don't personally care too much in this case, but something to think about in general).

**Comment by volth** [comment]
my consul service diverged from this one too much.

**Pull request closed without merging**